File: src/main/java/ttd/site/generator/SiteGeneratorConfigurationProperties.java
package ttd.site.generator;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.io.File;

/**
 * @author <a href="mailto:josh@joshlong.com">Josh Long</a>
 */
@Data
@ConfigurationProperties("site-generator")
public class SiteGeneratorConfigurationProperties {

	private File contentDirectory;

}

File: src/main/kotlin/ttd/site/generator/SiteGeneratorConfigurationProperties.kt
package ttd.site.generator

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.core.io.Resource

@ConstructorBinding
@ConfigurationProperties("site-generator")
data class SiteGeneratorConfigurationProperties(
    val commit: Boolean = true,
    val contentDirectory: Resource,
    val staticAssetsDirectory: Resource
)